### DISCUSSION TITLE
Singularity error

### DISCUSSION THREAD

**Discussion by saramoein372** [tuonglab/dandelion#117] [General]
Hi Kelvin,

I am running my Singularity again, but I get this error:

Formating fasta(s) : 0it [00:00, ?it/s]
Assigning genes : 0it [00:00, ?it/s]
Processing data file(s) : 0it [00:00, ?it/s]
Traceback (most recent call last):
  File "/share/dandelion_preprocess.py", line 177, in <module>
    main()
  File "/share/dandelion_preprocess.py", line 144, in main
    ddl.pp.reassign_alleles(samples,
  File "/opt/conda/envs/sc-dandelion-container/lib/python3.9/site-packages/dandelion/preprocessing/_preprocessing.py", line 1275, in reassign_alleles
    ] + [filepathlist_heavy[0]] + ['>'] + [
IndexError: list index out of range

Is there any way that I can remove this error? What can be the problem?

Thanks,
Sara

**Comment by saramoein372** [comment]
I also tried this:

 singularity run -B /athena/namlab/scratch/sam4032/BCR /athena/namlab/scratch/sam4032/BCR/sc-dandelion_latest.sif dandelion-preprocess
Formating fasta(s) :   0%|                                                                                                                                                                                                              | 0/11 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/share/dandelion_preprocess.py", line 177, in <module>
    main()
  File "/share/dandelion_preprocess.py", line 104, in main
    ddl.pp.format_fastas(
  File "/opt/conda/envs/sc-dandelion-container/lib/python3.9/site-packages/dandelion/preprocessing/_preprocessing.py", line 306, in format_fastas
    format_fasta(
  File "/opt/conda/envs/sc-dandelion-container/lib/python3.9/site-packages/dandelion/preprocessing/_preprocessing.py", line 84, in format_fasta
    raise OSError(
OSError: Path to fasta file is unknown. Please specify path to fasta file or folder containing fasta file. Starting folder should only contain 1 fasta file.


Again error of directory to fasta file

**Comment by liezeltamon** [comment]
Hi @zktuong,

Thanks for the tool! I've been trying it out to reannotate the genes using the container but I have been getting the IndexError as shown above. 

```
singularity run -B $PWD <dir>/dandelion_preprocess/sc-dandelion_latest.sif dandelion-preprocess
```
```
INFO:    Converting SIF file to temporary sandbox...
WARNING: underlay of /etc/localtime required more than 50 (91) bind mounts
Software versions:

dandelion==0.4.0 pandas==2.2.3 numpy==2.0.2 matplotlib==3.9.2 networkx==3.3 scipy==1.14.1

Begin preprocessing

command line parameters:
: --------------------------------------------------------------
    --meta = None
    --chain = ig
    --org = human
    --file_prefix = all
    --db = imgt
    --strain = None
    --sep = _
    --flavour = strict
    --filter_to_high_confidence = False
    --keep_trailing_hyphen_number = False
    --skip_format_header = False
    --skip_tigger = False
    --skip_reassign_dj = False
    --skip_correct_c = False
    --clean_output = False
: --------------------------------------------------------------

Formatting fasta(s) : |          | 0/0 [00:00<?, ?it/s]                                                                                                                   
Assigning genes : |          | 0/0 [00:00<?, ?it/s]                                                                                                                       
Processing data file(s) : |          | 0/0 [00:00<?, ?it/s]                                                                                                               
Traceback (most recent call last):
  File "/share/dandelion_preprocess.py", line 387, in <module>
    main()
  File "/share/dandelion_preprocess.py", line 318, in main
    ddl.pp.reassign_alleles(
  File "/opt/conda/envs/sc-dandelion-container/lib/python3.12/site-packages/dandelion/preprocessing/_preprocessing.py", line 1511, in reassign_alleles
    Path(filepathlist_heavy[0]),
         ~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
INFO:    Cleaning up image...
```

I am trying out with 1 sample as shown here:

```
.
├── all_contig_annotations.csv
└── all_contig.fasta
```

If I skip the tigger (see below),  I don't get the error but also no outputs. 

```
INFO:    Converting SIF file to temporary sandbox...
WARNING: underlay of /etc/localtime required more than 50 (91) bind mounts
Software versions:

dandelion==0.4.0 pandas==2.2.3 numpy==2.0.2 matplotlib==3.9.2 networkx==3.3 scipy==1.14.1

Begin preprocessing

command line parameters:
: --------------------------------------------------------------
    --meta = None
    --chain = ig
    --org = human
    --file_prefix = all
    --db = imgt
    --strain = None
    --sep = _
    --flavour = strict
    --filter_to_high_confidence = False
    --keep_trailing_hyphen_number = False
    --skip_format_header = False
    --skip_tigger = True
    --skip_reassign_dj = False
    --skip_correct_c = False
    --clean_output = False
: --------------------------------------------------------------

Formatting fasta(s) : |          | 0/0 [00:00<?, ?it/s]                                                                                                                   
Assigning genes : |          | 0/0 [00:00<?, ?it/s]                                                                                                                       
Assign isotypes 

Pre-processing finished.
 (0:00:00)
INFO:    Cleaning up image...
```
I think I'm just missing something basic here tbh, really sorry if that's the case!

Thanks in advance for the help!

Liezel
